app/controllers/katello/api/v2/products_bulk_actions_controller.rb

```diff
@@ -31,7 +31,7 @@ def sync_products
       validate_contents = ::Foreman::Cast.to_bool(params[:validate_contents])
 
       syncable_products = @products.syncable
-      syncable_repositories = Repository.where(:product_id => syncable_products).has_url
+      syncable_repositories = Repository.where(:product_id => syncable_products).has_url.in_default_view
```

Comment:
Note this seemed like a bug (and was breaking a test based on a fixture i added that shouldn't have been breaking it).